ReSharper.FSharp/src/FSharp.Psi.Services/src/Generate/GenerateProvider.fs

```diff
@@ -196,6 +197,22 @@ type FSharpOverridableMembersProvider() =
 type FSharpOverridingMembersBuilder() =
     inherit GeneratorBuilderBase<FSharpGeneratorContext>()
 
+    let addNewLineIfNeeded (typeDecl: IFSharpTypeDeclaration) (typeRepr: ITypeRepresentation) =
+        if isNull typeRepr || typeDecl.StartLine <> typeRepr.StartLine then () else
+
+        let currentIndent = typeRepr.Indent
+        let desiredIndent = typeDecl.Indent + typeDecl.GetIndentSize()
+        
+        use cookie = WriteLockCookie.Create(typeRepr.IsPhysical())
+        addNodesBefore typeRepr.FirstChild [
+            NewLine(typeRepr.GetLineEnding())
+            Whitespace(currentIndent)
+        ] |> ignore
+        
+        // to have good indents for begin/end keywords which might start out on different indents
```

Comment:
Could you please add a test showing this behaviour?

Reply:
`Anchor - Repr 07`
`Anchor - Repr 08`
are examples for this.
If we just do a
```shiftNode -(currentIndent - desiredIndent) typeRepr```
for them, the `end` keyword is shifted too much to the left.

Reply:
I see, nice, thanks!